### ISSUE TITLE
Can't record a new payment

### ISSUE THREAD

**Issue by dev-allinappli** [Project60/org.project60.sepa#735]
When trying to register I'm getting this error:

` 8000: Couldn't create SEPA mandate. Error was: DB Error: already exists`

Checking CiviCRM's log I can find the related entries:
```

[debug] SepaPP: createPendingMandate had nothing to do

2024-12-19 21:38:46+0100  [debug] SepaPP: createPendingMandate for contribution ID []

2024-12-19 21:38:46+0100  [debug] SepaPP: getPendingContributionID couldn't find a contribution ID.

2024-12-19 21:38:46+0100  [debug] SepaPP: createPendingMandate had nothing to do

2024-12-19 21:38:46+0100  [warning] Deprecated function doDirectPayment, use doPayment.
CRM_Core_Error::deprecatedFunctionWarning
CRM_Core_Payment::doPayment
CRM_Contribute_Form_Contribution_Confirm::postProcessMembership
CRM_Contribute_Form_Contribution_Confirm::processMembership
Array
(
    [civi.tag] => deprecated
)

2024-12-19 21:38:46+0100  [warning] Deprecated function CRM_Core_Payment::doPayment, use payment processors should throw exceptions rather than return errors.
CRM_Core_Error::deprecatedFunctionWarning
CRM_Core_Payment::doPayment
CRM_Contribute_Form_Contribution_Confirm::postProcessMembership
CRM_Contribute_Form_Contribution_Confirm::processMembership
Array
(
    [civi.tag] => deprecated
)

2024-12-19 21:38:46+0100  [debug] org.project60.sepa: get_parameter for unknown key: pp_buffer_days
```
It's a bit inconsistent because it happens to work from times to times and server log and web console are also not very helpful.
We're using version 1.11.0 with SepaPP 1.3-dev on 5.71.1

**Comment by peth-systopia** [comment]
Hi, 
afaik we have not encountered this issue on systems we maintain, and we generally can not provide free support on this if it happens in your set-up. If you want us to work on your issue in terms of paid support, feel free to get in touch via info@systopia.de. 
Please also note that the [SepaPP extension](https://github.com/Project60/org.project60.sepapp) which is part of your set-up is no longer maintained by us (SYSTOPIA).

**Comment by bjendres** [comment]
> Please also note that the [SepaPP extension](https://github.com/Project60/org.project60.sepapp) which is part of your set-up is no longer maintained by us (SYSTOPIA).

See https://github.com/Project60/org.project60.sepapp/issues/36

**Comment by Detsieber** [comment]
@allinappliadmin This clearly seems related to the Extension sepapp, which contains the SEPA payment processor.

I would recommend:
1. Better do not use the "master" code (which is called "1.3-dev" for some reason). Better use the code tagged with [1.2-beta4](https://github.com/Project60/org.project60.sepapp/releases/tag/1.2-beta4).
2.Use the SEPA "NG" payment processor.
3. I this would not help, please file an issue in the repository of sepapp: https://github.com/Project60/org.project60.sepapp/issues

**Comment by wintermoor** [comment]
Does the debug info "get_parameter for unknown key: pp_buffer_days" refer to https://github.com/Project60/org.project60.sepapp/issues/2 ?

**Comment by bjendres** [comment]
> Does the debug info "get_parameter for unknown key: pp_buffer_days" refer to [Project60/org.project60.sepapp#2](https://github.com/Project60/org.project60.sepapp/issues/2) ?

I don't think so, ``pp_buffer_days`` is used by ``org.project.sepa`` as well.

**Comment by vurt2** [comment]
@wintermoor I think it comes from sepaPP... you could get rid of it if you insert "pp_buffer_days" with a value of "0" into the settings table like in the ticket you refered to.

**Comment by bjendres** [comment]
>  you could get rid of it if you insert "pp_buffer_days" with a value of "0" into the settings table like in the ticket you refered to.

Sure, but the easiest option would be to set the "Buffer Days" in the CiviSEPA settings to whatever value you want.

**Comment by bjendres** [comment]
> [@bjendres](https://github.com/bjendres) You are correct - the settings code is in sepa, but in this case it is called from sepaPP. And using the interface is better than a db insert ;)

Ok, great. But I'd assume your original problem is still not solved, right? 

I think you should re-file this issue here: https://github.com/Project60/org.project60.sepapp/issues - this is the payment processor you're using, right?

**Comment by wintermoor** [comment]
This is surely an issue for sepapp. 
My buffer day is set to 0 in CiviSEPA config. My log shows "[debug] org.project60.**sepa**: get_parameter for unknown key: pp_buffer_days" nevertheless. But again: buffer days aren't topic of this issue, sorry.

**Comment by vurt2** [comment]
@wintermoor I just fixed this in sepapp. @bjendres Thank you for reminding me - I think this issue can be closed as this all seems more sepapp related...
